File: wesafe.net/WeSafe.DAL.Entities/DetectedCamera.cs
using WeSafe.Shared.Enumerations;

namespace WeSafe.DAL.Entities
{
    public class DetectedCamera
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Ip { get; set; }

        public string Port { get; set; }

        public string Login { get; set; }

        public string Password { get; set; }

        public int DeviceId { get; set; }

        public Device Device { get; set; }

        public CameraState State { get; set; }

        public string DetectingMethod { get; set; }

        public string ConnectFailureText { get; set; }
    }
}

File: wesafe.net/WeSafe.Web/ClientApp/src/app/models/detected-camera.ts
export interface IDetectedCamera {
    id: number;
    name: string;
    ip: string;
    port: string;
    login: string;
    password: string;
    deviceId: number;
    state: number;
    detectingMethod: string;
    connectFailureText: string;
}

export class ConnectingDetectedCamera {
    id: number;
    login: string;
    password: string;
}
